presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoResultSet.java

```diff
@@ -1807,23 +1808,25 @@ private static Number toNumber(Object value)
         private final StatementClient client;
         private final BlockingQueue<T> rowQueue = new ArrayBlockingQueue<>(MAX_QUEUED_ROWS);
         private final CompletableFuture<Void> future;
-        private Thread parent;
+        private final Semaphore semaphore = new Semaphore(0);
```

Comment:
move next to `rowQueue` and document semaphore meaning with a code comment (ie what does a permit represent)